src/main/java/Chatbot/TaskStorage.java

```diff
@@ -0,0 +1,124 @@
+package Chatbot;
+
+import Tasks.Task;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The {@code TaskStorage} class manages the storage and retrieval of tasks.
+ * It utilizes {@code HarddiskStorage} for persistent storage and maintains a map of tasks in memory.
+ * The class supports adding, marking, retrieving, deleting, listing, and finding tasks.
+ */
+public class TaskStorage {
```

Comment:
I like the clear descriptions provided in the javadoc throughout the project, it makes it easy to understand what the various methods do